### PR TITLE
Feature add home banner

### PR DESCRIPTION
**Pull request by Melissa-Martinez-XL**
## Ticket

* [Add top banner to main page](https://www.notion.so/xmartlabs/Add-top-banner-to-main-page-93a9bb3bac73463e92cbaf9b5cee5933)

## Type of change

* [ ] Fix
* [X] Story
* [ ] Chore

## Description of the change

I have added the home banner and created a communication with the other banners to change the image on the pages. 

## Screenshot/Execution

Captura desde 2022-12-15 11-07-36
Captura desde 2022-12-15 11-08-16

## Related PRs
(https://github.com/xmartlabs/xl-blog/pull/24)

### CODE REVIEW

File: src/components/card/card.jsx [hunk 1]
```diff
@@ -8,15 +8,17 @@ import { Category } from "../category";
...
-const Card = ({ data }) => (
-  <article key={data.id} className={cardStyles.container}>
-    <img className={cardStyles.styledImage} src={data.frontmatter.thumbnail}/>
-    <Category data={data.frontmatter.category}/>
-    <Link className={cardStyles.styledLink} to={`/${_.kebabCase(data.frontmatter.permalink)}`}>
-      {data.frontmatter.title}
-    </Link>
-  </article>
-);
+const Card = ({ data }) => {
```
Comment (diegorosolino) [inline src/components/card/card.jsx:11]: Why this change?

File: src/components/home-banner/home-banner.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from "react"
+
+import { StarIcon } from "../star-icon/star-icon";
+import { PolygonIcon } from "../polygon-icon/polygon-icon";
+
```
Comment (diegorosolino) [inline src/components/home-banner/home-banner.jsx:5]: Maybe you can define an icon folder and put these components into this folder

File: src/components/home-banner/home-banner.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import * as styles from "./home-banner.module.scss";
+
+const HomeBanner = () => (
+  <div className={styles.containerTitleHome}>
+    <h1 className="text__heading__one__big__black">Xmartlabs Blog</h1>
+    <div className={styles.textContainer}>
+      <h2 className="text__heading__two__big__black">
+        <StarIcon />
```
Comment (diegorosolino) [inline src/components/home-banner/home-banner.jsx:13]: Weren't you going to remove this icon?

File: src/components/home-banner/home-banner.module.scss [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+}
+
+.polygonContainer {
+  display: flex;
+  justify-content: center;
+  align-items: end;
+  position: absolute;
+  width: 82%;
+  height: 62%;
```
Comment (diegorosolino) [inline src/components/home-banner/home-banner.module.scss:27]: Are you sure that we need these props?

File: src/components/polygon-icon/polygon-icon.jsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+export const PolygonIcon = ({ className }) => (
```
Comment (diegorosolino) [inline src/components/polygon-icon/polygon-icon.jsx:3]: Please add PropTypes and defaultValues for this component

File: src/layouts/layout.jsx [hunk 6]
```diff
@@ -150,4 +160,9 @@ function Layout({ children }) {
   );
 }
 
+Layout.propTypes = {
+  children: PropTypes.object, 
+  type: BannerType,
```
Comment (diegorosolino) [inline src/layouts/layout.jsx:165]: Please remove this value

File: src/layouts/layout.jsx [hunk 7]
```diff
@@ -150,4 +160,9 @@ function Layout({ children }) {
   );
 }
 
+Layout.propTypes = {
```
Comment (diegorosolino) [inline src/layouts/layout.jsx:163]: Please add defaultValues for this component

File: src/layouts/layout.jsx [hunk 8]
```diff
@@ -54,34 +59,39 @@ export const StyledFooterText = styled.label`
...
+                <NavMenu />
+              </div>
+              <StyledGetStartedButton id="header-getintouch" 
+                href="#/" 
+              >
+              <StyledGetStartedTextButton>GET STARTED</StyledGetStartedTextButton>
+              </StyledGetStartedButton>
+            </StyledContainerHeader>
+          </StyledContainerNavBarXL>
+          {contextState === BannerType.home && <HomeBanner />}
```
Comment (diegorosolino) [inline src/layouts/layout.jsx:91]: why this?

File: src/components/card/card.jsx (hunk 1 above)
Comment (Melissa-Martinez-XL) [reply src/components/card/card.jsx:11]: I thought that since there is no function or React Hook and also the only thing it returns is html code, in addition to a component, it was correct to put the parenthesis. But I can change it.

Comment (Melissa-Martinez-XL) [reply src/components/card/card.jsx:11]: Because I was testing some functions and hooks, but in the end I didn't use them, so I forgot to change the curly braces again.

File: src/components/home-banner/home-banner.jsx (hunk 2 above)
Comment (Melissa-Martinez-XL) [reply src/components/home-banner/home-banner.jsx:5]: Done

File: src/components/home-banner/home-banner.jsx (hunk 3 above)
Comment (Melissa-Martinez-XL) [reply src/components/home-banner/home-banner.jsx:13]: No, only in the mobile version.

File: src/components/home-banner/home-banner.module.scss (hunk 4 above)
Comment (Melissa-Martinez-XL) [reply src/components/home-banner/home-banner.module.scss:27]: I am using them to position the polygons, I can change them for left and top.

File: src/layouts/layout.jsx (hunk 6 above)
Comment (Melissa-Martinez-XL) [reply src/layouts/layout.jsx:165]: The whole code, or which specific value?

File: src/components/home-banner/home-banner.jsx (hunk 3 above)
Comment (diegorosolino) [reply src/components/home-banner/home-banner.jsx:13]: And do you remove it on mobile view?

File: src/layouts/layout.jsx [hunk 9]
```diff
@@ -54,34 +59,39 @@ export const StyledFooterText = styled.label`
...
-              href="#/" 
-            >
-            <StyledGetStartedTextButton>GET STARTED</StyledGetStartedTextButton>
-            </StyledGetStartedButton>
-          </StyledContainerHeader>
-        </StyledContainerNavBarXL>
-        {children}
-      </div>
+      <AppContext.Provider value={{ state: contextState, setState: setContextState }}>
+        <div className={classnames(styles[`${contextState}Banner`])}>
```
Comment (diegorosolino) [inline src/layouts/layout.jsx:68]: You don't need `classnames`
```suggestion
        <div className={styles[`${contextState}Banner`]}>
```

File: src/layouts/layout.jsx (hunk 6 above)
Comment (diegorosolino) [reply src/layouts/layout.jsx:165]: I was talking about the type prop, you don't use this prop, right?

File: src/components/card/card.jsx [hunk 10]
```diff
@@ -12,7 +12,7 @@ const Card = ({ data }) => (
   <article key={data.id} className={cardStyles.container}>
     <img className={cardStyles.styledImage} src={data.frontmatter.thumbnail}/>
     <Category data={data.frontmatter.category}/>
-    <Link className={cardStyles.styledLink} to={`/${_.kebabCase(data.frontmatter.permalink)}`}>
```
Comment (diegorosolino) [inline src/components/card/card.jsx:15]: Remove this empty space

File: src/layouts/layout.jsx (hunk 8 above)
Comment (diegorosolino) [reply src/layouts/layout.jsx:91]: Why not use the HomeBanner in the component that renders as children?

File: src/layouts/layout.jsx (hunk 9 above)
Comment (Melissa-Martinez-XL) [reply src/layouts/layout.jsx:68]: Done

File: src/components/card/card.jsx (hunk 10 above)
Comment (Melissa-Martinez-XL) [reply src/components/card/card.jsx:15]: Done

File: src/components/home-banner/home-banner.jsx (hunk 3 above)
Comment (diegorosolino) [reply src/components/home-banner/home-banner.jsx:13]: And do you remove it on mobile view?

Comment (diegorosolino) [reply src/components/home-banner/home-banner.jsx:13]: And do you remove it on mobile view?

**Review by diegorosolino — APPROVED** [review]
